src/components/ApiReference/ReferenceSection.js

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import styled from "styled-components";
+import PropTypes from "prop-types";
+import { MDXRenderer } from "gatsby-plugin-mdx";
+
+import { apiReferenceComponents } from "constants/docsComponentMapping";
+import { PALETTE } from "constants/styles";
+
+import { HorizontalRule } from "basics/Text";
+import { EditIcon } from "basics/Icons";
+import { Link } from "basics/Links";
+
+import { Route } from "components/ApiReference/Route";
+import { CustomColumn, NestedRow } from "components/ApiReference/SharedStyles";
+import { TrackedContent } from "components/SideNav";
+import { useLocation } from "@reach/router";
+import { Context as ScrollRouterContext } from "./ScrollRouter";
+
+const { h1: H1, h2: H2 } = apiReferenceComponents;
+
+const MainEl = styled.main`
+  display: block;
+`;
+const ArticleEl = styled.article`
+  display: block;
+  &:first-child {
+    margin-top: 5rem;
+  }
+`;
+
+const Article = ({ children, path }) => {
+  // The scroll router overrules Reach Router once the page loads, but we need
+  // Reach router's location to get initial path
+  const { pathname } = useLocation();
+  const [isActive, setIsActive] = React.useState(pathname);
+  const { subscribe } = React.useContext(ScrollRouterContext);
+
+  React.useEffect(
+    () =>
+      subscribe((newPath) => {
+        setIsActive(path === newPath);
+      }),
+    [subscribe, path],
+  );
+
+  return (
+    <MainEl name={path} hidden={!isActive}>
```

Comment:
I added a `<main>` element wrapping each section. There's only supposed to be 1 main element on a page, so it also adds a `hidden` attribute if the current section doesn't match the current URL. And this is maybe a bit of magical thinking, but it also adds the current URL as the `name` attr, cuz hey maybe that'll help?